Add CLARA Clustering algorithm (#83)
scikit-learn-contrib/scikit-learn-extra: doc/modules/cluster.rst

    for performing face recognition. International Journal of Soft Computing,
    Mathematics and Control, 3(3), pp 1-12.

CLARA
=====

    :class:`CLARA` is related to the :class:`KMedoids` algorithm. CLARA

>     (Clustering for Large Applications) extends k-medoids approach for a
>     large number of objects. This algorithm uses a sampling approach.

Reviewer: Maybe,
```suggestion
    (Clustering for Large Applications) extends k-medoids to be more scalable,
    uses a sampling approach.
```

    .. topic:: Examples:

      * :ref:`sphx_glr_auto_examples_plot_clara_digits.py`: Applying K-Medoids on digits
        with various distance metrics.

    **Algorithm description:**

>     CLARA use `sample` random samples of the dataset, each of size `sampling_size`

Reviewer: ```suggestion
    CLARA uses random samples of the dataset, each of size `sampling_size`
```
?
